### ISSUE TITLE
captum insight on bert model can not works

### ISSUE THREAD

**Issue by dupf** [pytorch/captum#803]
The following is my code about captum insight on bert mode, it doesn't works, should you supply examples about captum insight on bert model. 

import os
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import torch.nn.functional as F

import torch
import torch.nn as nn

import sys

sys.path.append('/data/fakenews/fakefusion/experiment_analysis/captum-master/')

from transformers import (
    AutoConfig,
    AutoModelForSequenceClassification,
    AutoTokenizer,
    DataCollatorWithPadding,
    EvalPrediction,
    HfArgumentParser,
    PretrainedConfig,
    Trainer,
    TrainingArguments,
    default_data_collator,
    set_seed,
)

from captum.attr import visualization as viz
from captum.attr import IntegratedGradients, LayerConductance, LayerIntegratedGradients, LayerActivation
from captum.attr import configure_interpretable_embedding_layer, remove_interpretable_embedding_layer
from captum.insights.attr_vis.features import ImageFeature, TextFeature
from captum.insights import AttributionVisualizer, Batch
import datasets
import numpy as np

from captum.attr import TokenReferenceBase, configure_interpretable_embedding_layer, remove_interpretable_embedding_layer
device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

model_path ="/data/fakenews/fakefusion/code/model_albert_v7"
model_fake = AutoModelForSequenceClassification.from_pretrained(
        model_path,
        config=model_path

    )
model_fake.to(device)
tokenizer = AutoTokenizer.from_pretrained(model_path)

def predict(inputs, token_type_ids=None, position_ids=None, attention_mask=None):
    output = model_fake(inputs, token_type_ids=token_type_ids,
                 position_ids=position_ids, attention_mask=attention_mask, )
    
    return output

ref_token_id = tokenizer.pad_token_id # A token used for generating token reference
sep_token_id = tokenizer.sep_token_id # A token used as a separator between question and text and it is also added to the end of the text.
cls_token_id = tokenizer.cls_token_id # A token used for prepending to the concatenated question-text word sequence
def construct_input( text, ref_token_id, sep_token_id, cls_token_id):
    text_ids = tokenizer.encode(text, add_special_tokens=False)
    input_ids = [cls_token_id]  + text_ids + [sep_token_id]
    ref_input_ids = [cls_token_id] +  [ref_token_id] * len(text_ids) + [sep_token_id]
    return torch.tensor([input_ids], device=device), torch.tensor([ref_input_ids], device=device), 

def construct_input_token_type(input_ids, sep_ind=0):
    seq_len = input_ids.size(1)
    token_type_ids = torch.tensor([1 for i in range(seq_len)], device=device)
    ref_token_type_ids = torch.zeros_like(token_type_ids, device=device)# * -1
    return token_type_ids, ref_token_type_ids
def construct_input_pos_id(input_ids):
    seq_length = input_ids.size(1)
    position_ids = torch.arange(seq_length, dtype=torch.long, device=device)
    ref_position_ids = torch.zeros(seq_length, dtype=torch.long, device=device)
    position_ids = position_ids.unsqueeze(0).expand_as(input_ids)
    ref_position_ids = ref_position_ids.unsqueeze(0).expand_as(input_ids)
    return position_ids, ref_position_ids
    
def construct_attention_mask(input_ids):
    return torch.ones_like(input_ids)

def encode_text(text):
    """ Turn a question into a vector of indices and a question length """
    question_arr = text
    vec = torch.zeros(len(question_arr), device=device).long()
    text_ids,_ = construct_input(text, ref_token_id, sep_token_id, cls_token_id)  #tokenizer.encode(text, add_special_tokens=False)    
    token_type_ids,_ = construct_input_token_type(text_ids)
    position_ids, ref_position_ids = construct_input_pos_id(text_ids)
    attention_mask = construct_attention_mask(position_ids)
    return text_ids, token_type_ids, position_ids, attention_mask, torch.tensor(text_ids.size(), device=device)

interpretable_embedding = configure_interpretable_embedding_layer(
    model_fake, "albert.embeddings"
)

PAD_IND = tokenizer.pad_token_id
token_reference = TokenReferenceBase(reference_token_idx=PAD_IND)

def fakenews_dataset(texts, targets):
    for text, target in zip(texts, targets):
        input_ids, token_type_ids, position_ids, attention_mask, q_len = encode_text(text)
        q_len = q_len.unsqueeze(0)
        target_idx = torch.tensor(targets) #answer_to_index[target
  
        yield Batch(
            inputs=(input_ids), labels=(target_idx) )
        #additional_args=(attention_mask))

def baseline_text(x):
    seq_len = x.size(0)  
    ref_input_ids = [cls_token_id] +  [ref_token_id] * (seq_len-2) + [sep_token_id]
    print("========== output ===========", ref_input_ids)
    input_ref_embed = torch.tensor([ref_input_ids], device=device)
    return input_ref_embed #interpretable_embedding.indices_to_embeddings(input_ref_embed).squeeze(0)

def input_text_transform(x):
    print("==== output embedding =======", x)

    return x #out_emb.squeeze(0) #interpretable_embedding.indices_to_embeddings(x)

def itos(input):  
    print("=======input input========", input)
    return input #[question_words[int(i)] for i in input.squeeze(0)]

def score_func(o):
    print("output ======", o)
    return F.softmax(o[0], dim=1)

datasetfk = fakenews_dataset(
    ["This is example "], 
    [1]
    )

features =[
        TextFeature(
        "TextInput",
        input_transforms=[input_text_transform],
        baseline_transforms=[baseline_text],
        visualization_transform=None,
           )
         ]
def get_classes():
    classes = [
        "real",
        "fake"
    ]
    return classes

visualizer = AttributionVisualizer(
    models = [model_fake],
    score_func = score_func,
    features = features,
    dataset = datasetfk,
    classes=get_classes(),
)

remove_interpretable_embedding_layer(model_fake, interpretable_embedding)
visualizer.serve(port=1772,bind_all=True, debug=True)

**Comment by bilalsal** [comment]
Hi @dupf , could you elaborate a bit?

Are you running your code within a notebook? 
What happens when you run visualizer.serve()?
Are there any backend errors (in Python) or frontend errors (browser errors you can see in the "inspect" panel of your browser)?
